_posts/2020-04-28-PC-Engines-Firmware-v4-11-0-6.md

```diff
@@ -0,0 +1,210 @@
+---
+layout: post
+title:  "PC Engines apu coreboot Open Source Firmware v4.11.0.6"
+date:   2020-03-31
+categories: Firmware
+---
+# PC Engines apu coreboot Open Source Firmware v4.11.0.6
+
+## Key changes
+
+Mainline:
+
+1. **Rebased** with official coreboot repository **commit** d6f7ec5.
+2. **Updated sortbootorder** to **v4.6.18** bringing the PCI Express power
+   management features runtime option. For details refer to
+   [sortbootorder documentation](https://github.com/pcengines/sortbootorder#settings-description).
+   When PCI Express power management features features are enabled, the network
+   controllers (NICs and WIFi cards) may have reduced performance at the cost
+   of reduced power consumption. By default this option will be disabled to not
+   [impact the network performance](https://github.com/pcengines/coreboot/issues/387).
+3. **Reverted changes to ACPI CPU definitions** causing BSD systems to
+   [not probe CPU frequency driver](https://github.com/pcengines/coreboot/issues/389).
+   **The ACPI compliance of current BSD systems is not up to date**, The
+   situation should improve when the distribution will start to use FreeBSD
+   12.x, which works well with most recent rules of defining processors in
+   ACPI.
+4. **Reverted changes with PCIe reset logic** causing
+   [mPCIe2 slot connected modules to not appear in OS](https://github.com/pcengines/coreboot/issues/388).
+   The change did more harm than good. We are working to improve the PCIe
+   modules detection in firmware, which is dependent on the AGESA.
+5. **Added IOMMU IVRS generation expanded with IVHD type 11h for newer Xen**.
+   This change should allow newer Xen images to utilize more IOMMU features.
+6. **Fixed memtest hang on apu1**.
+7. **Fixed TPM2 detection on FreeBSD 12.1**. Since FreeBSD 12.1 the TPM2
+   support is available along with FreeBSD ports offering TPM2 tools. We will
+   provide documentation how to install and utilize those tools on FreeBSD
+   systems soon.
+
+## coreboot community
+
+Patches merged by community:
+
+* [drivers/pc80/tpm/tis.c: change the _HID and _CID for TPM2 device](https://review.coreboot.org/c/coreboot/+/39699)
+* [arch/x86/acpi: add definitions for IVHD type 11h](https://review.coreboot.org/c/coreboot/+/40041)
+* [nb/amd/pi/00730F01/northbridge.c: refactor IVRS generation](https://review.coreboot.org/c/coreboot/+/40042)
+* [Revert "mb/pcengines/apu2: add reset logic for PCIe slots"](https://review.coreboot.org/c/coreboot/+/40147)
+
+**Total:**
+
+* 308 lines added,
+* 245 lines removed,
+
+in official coreboot repository.
+
+## Other news
```

Comment:
@pietrushnic I came up with an idea to add another section which describes TrenchBoot related effort to the newsletter. Please have a look at this section and review, add suggestions, etc., if any.

Reply:
@miczyg1 very good idea IMO